src/lib.rs

```diff
@@ -44,14 +54,88 @@ impl Reading {
         humidity: String,
         pressure: String,
         dust_concentration: String,
-        air_purity: String
+        air_purity: String,
     ) -> Reading {
         Reading {
             temperature,
             humidity,
             pressure,
             dust_concentration,
-            air_purity
+            air_purity,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+struct ResponseLog<'a, W: Write> {
+    debug: bool,
+    writer: &'a mut W,
+}
+
+impl<'a, W: Write> ResponseLog<'a, W> {
+    pub fn new(debug: bool, writer: &'a mut W) -> Self {
```

Comment:
Have you considered a static lifetime that is also trait bound? Here's an example of what I mean: https://doc.rust-lang.org/rust-by-example/scope/lifetime/static_lifetime.html#trait-bound